cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -49,6 +49,24 @@
   void addPartition(PartitionKey key, String path);
 
   /**
+   * Add a partition for a given partition key, stored at a given path (relative to the file set's base path),
+   * with the given metadata.
+   */
+  void addPartition(PartitionKey key, String path, Map<String, String> metadata);
+
+  /**
+   * Adds a new metadata entry for a particular partition.
+   * Note that existing entries can not be updated.
+   */
+  void addMetadata(PartitionKey key, String metadataKey, String metadataValue);
+
+  /**
+   * Adds a set of new metadata entries for a particular partition
+   * Note that existing entries can not be updated.
```

Comment:
Does that mean it will throw exception from this method or get ignored silently?